cypress/e2e/ConfigChangeRegression.cy.ts

```diff
@@ -0,0 +1,129 @@
+    ).as('coolAssessment');
+    cy.intercept(
+      'GET',
+      'https://raw.githubusercontent.com/neurobagel/communities/main/configs/Cool/diagnosis.json',
+      { body: defaultDiagnosisTerms }
+    ).as('coolDiagnosis');
+    cy.intercept(
+      'GET',
+      'https://raw.githubusercontent.com/neurobagel/communities/main/configs/Cool/sex.json',
+      { body: defaultSexTerms }
+    ).as('coolSex');
+  });
+
+  it('Should update assessment vocabulary after changing configs', () => {
+    cy.visit('http://localhost:5173');
+    cy.get('[data-cy="next-button"]').click();
+
+    // Upload view - Select Neurobagel config
+    cy.wait('@configList');
+    cy.get('[data-cy="config-card-dropdown"] input').should('have.value', 'Neurobagel');
+    cy.get('[data-config-loading="false"] ').should('exist');
+
+    cy.get('[data-cy="datatable-upload-input"]').selectFile(mockDataTableFilePath, {
+      force: true,
+    });
+    cy.get('[data-cy="next-button"]').click();
+
+    // Column Annotation view - Annotate a column as Assessment Tool with the default Neurobagel config
+    cy.get('[data-cy="column-annotation-container"]').should('be.visible');
+    cy.get('[data-cy="4-column-annotation-card-standardized-variable-dropdown"]').type(
+      'assessment{downArrow}{enter}'
+    );
+    cy.get('[data-cy="next-button"]').click();
+
+    // Multi-Column Measures view - Move to Multi-Column Measures, then backtrack to the Upload view to change configs
+    cy.get('[data-cy="multi-column-measures-tab-Assessment Tool"]').should('be.visible');
+    cy.get('[data-cy="add-term-card-button"]').click();
+    cy.get('[data-cy="multi-column-measures-card-0-title-dropdown"]').click();
+    cy.get('ul[role="listbox"]').should('contain', 'Robson Ten Group Classification System');
+    cy.get('[data-cy="back-button"]').click();
+    cy.get('[data-cy="back-button"]').click();
+
+    // Upload view - Change to "Cool" config and wait for its vocab to load
+    cy.get('[data-cy="config-card-dropdown"]').type('cool{downArrow}{enter}');
+    cy.wait(['@coolConfig', '@coolAssessment', '@coolDiagnosis', '@coolSex']);
+
+    // Re-run annotation using the new config to ensure we see the refreshed vocab
+    cy.get('[data-cy="next-button"]').click();
+    cy.get('[data-cy="column-annotation-container"]').should('be.visible');
+    cy.get('[data-cy="4-column-annotation-card-standardized-variable-dropdown"]').type(
+      'Assessment{downArrow}{enter}'
+    );
+    cy.get('[data-cy="next-button"]').click();
+
+    // New assessment terms from "Cool" should replace the Neurobagel list
+    cy.get('[data-cy="multi-column-measures-tab-Assessment Tool"]').should('be.visible');
+    cy.get('[data-cy="add-term-card-button"]').click();
+    cy.get('[data-cy="multi-column-measures-card-0-title-dropdown"]').click();
+    cy.get('ul[role="listbox"]').should('contain', 'Cool Name');
+    cy.get('ul[role="listbox"]').should('not.contain', 'Robson Ten Group Classification System');
```

Comment:
I think it would be cleaner if you mocked both assessment vocabularies in this test so that you can transparently make the assertion here (i.e. "can't see oldName any more, but can see newName"). Will also help not to break the test if we update our (mocked) assessment terms